### PR TITLE
Document contribution guidelines

### LINKED ISSUE #1484: contributing guidelines lack development instructions
The [contributing guidelines](https://github.com/go-swagger/go-swagger/blob/master/.github/CONTRIBUTING.md) don't give a lot of guidance for building or testing the project.

I've tried running some make commands on a fresh checkout and get a lot of errors
```bash
$ make generate
make: *** No rule to make target `build/image_build', needed by `generate'.  Stop.

$ make test
...
cmd/swagger/commands/generate/shared.go:61::warning: declaration of "err" shadows declaration at cmd/swagger/commands/generate/shared.go:46 (vetshadow)
cmd/swagger/commands/generate/shared.go:67::warning: declaration of "err" shadows declaration at cmd/swagger/commands/generate/shared.go:46 (vetshadow)
cmd/swagger/swagger.go:133::warning: Expect file permissions to be 0600 or less,MEDIUM,HIGH (gas)
cmd/swagger/commands/generate/model_test.go:37:22:warning: error return value not checked (defer os.RemoveAll(generated)) (errcheck)
cmd/swagger/commands/generate/model_test.go:39:15:warning: error return value not checked (flags.Parse(m)) (errcheck)
cmd/swagger/commands/generate/shared.go:26:2:warning: unused struct field github.com/go-swagger/go-swagger/cmd/swagger/commands/generate.shared.genOpts (structcheck)
cmd/swagger/swagger.go:32:1:warning: Debug is unused (deadcode)
cmd/swagger/commands/generate/shared.go:26:2:warning: field genOpts is unused (U1000) (megacheck)
cmd/swagger/swagger.go:34:2:warning: var Debug is unused (U1000) (megacheck)
```

Some "getting started" section with a few bash commands would help. You could even just link to CI's build instructions.

### PR DESCRIPTION
**Pull request by fredbi**
- Fixes #1484

### CODE REVIEW

File: docs/guidelines/README.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+##### Testing PRs requiring integration of a dependency (e.g. another pending PR on `go-openapi`)
+
+With `go-swagger` (vendored):
+- prepare a "WIP" PR with a temporary vendor update commit
+- this vendor update temporarily alters the `branch` (or `source` for forks) in `Gopkg.toml`
+to get the proper version of the required dependency
+
+With `go-openapi` (non vendored):
+- for your VIP PR, temporarily alter the CI config script (e.g. `.travis.yml`) and
```
Comment (zhuangqh) [inline docs/guidelines/README.md]: s/VIP/WIP ? @fredbi

Comment (fredbi) [reply docs/guidelines/README.md]: Ah grmbl yes you're right

**Comment by zhuangqh** [comment]
LGTM

**Comment by fredbi** [comment]
ok fixed this typo (and a few others)

**Pull request merged**